### PR TITLE
Add better messages around LookupCoordinatorManager failures

### PR DESCRIPTION
**Pull request by drcrallen**
* Catches #3026 and other LookupCoordinatorManager failures

### CODE REVIEW

**Comment by drcrallen** [comment]
Closing and restarting since dependency was merged in

File: server/src/test/java/io/druid/server/lookup/cache/LookupCoordinatorManagerTest.java [hunk 1]
```diff
@@ -1245,6 +1254,8 @@ public void testStartTooMuch() throws Exception
         lookupCoordinatorManagerConfig
     );
     manager.start();
+    Assert.assertTrue(manager.backgroundManagerIsRunning());
+    Assert.assertFalse(manager.waitForBackgroundTermination(10));
```
Comment (nishantmonu51) [inline server/src/test/java/io/druid/server/lookup/cache/LookupCoordinatorManagerTest.java]: minor nit: could we also test post stop semantics, i.e backgroundManagerIsRunning is false & waitForBackgroundTermination returning true

**Comment by nishantmonu51** [comment]
👍

File: server/src/test/java/io/druid/server/lookup/cache/LookupCoordinatorManagerTest.java (hunk 1 above)
Comment (drcrallen) [inline server/src/test/java/io/druid/server/lookup/cache/LookupCoordinatorManagerTest.java]: Good call, added.

**Comment by b-slim** [comment]
👍

**Pull request merged**